src/dma.rs

```diff
@@ -142,7 +146,11 @@ impl<B, C: Channel, T> TransferInner<B, C, T> {
 /// The implementing type must be safe to use for DMA reads. This means:
 ///
 /// - It must be a pointer that references the actual buffer.
-/// - The requirements documented on `read_buffer` must be fulfilled.
+/// - The following requirements must be fulfilled by `read_buffer`:
+///   - The function must always return the same values, if called multiple
+///     times.
```

Comment:
I don't think this is true in the usual `StableDeref` sense of things.

Reply:
Yeah, I also had issues with that. The `{Read,Write}Buffers` demand that the buffer we give out remains stable and valid as long as the object lives. But that is not true, when we use `StableDeref`. With `StableDeref` it is only true if we also demand that `&mut self` is not called.

Ideally, we would just drop the use of `StableDeref` and be strict on the initial demand. That would keep the safety guarantees relatively simple with few edge cases. Also the `{read,write}_buffer` methods wouldn't need to be `unsafe` anymore. BUT that would also prevent us from using `Vec` and `Box` for DMA. Those could only safely implement the traits if they were wrapped in a newtype that makes the `&mut self` methods inaccessible. Which is inconvenient. And shouldn't be necessary, since `Vec` and `Box` are perfectly fine for DMA...

So... I'll update those `Safety` comments to also include the `&mut self` edge case. But maybe we can think of something better at some point.